src/pushsource/_impl/model/ami.py

```diff
@@ -99,6 +99,21 @@ class AmiPushItem(VMIPushItem):
     )
     """Billing codes associated with this image."""
 
+    description = attr.ib(type=str, default=None, validator=optional_str)
+    """A description for the image"""
+
+    release_notes = attr.ib(type=str, default=None, validator=optional_str)
+    """"Release notes for AMI """
+
+    usage_instructions = attr.ib(type=str, default=None, validator=optional_str)
+    """Instructions for use of AMI"""
+
+    recommended_instance_type = attr.ib(type=str, default=None, validator=optional_str)
+    """Recommended instance type for AMI"""
+
+    product_type = attr.ib(type=str, default=None, validator=optional_str)
+    """Type of product e.g. "AMIProduct", "SaasProduct", "ServerProduct". """
```

Comment:
Where does this list of types come from?

Reply:
At the moment, AWS doesn't have any list for these product types. I have asked them to include them in the docs in a meeting but they're not added ATM. So to get these I've just used the Marketplace webpage and watched the API calls for the differing product types.

Reply:
So it's defined on the AWS side, not something we've made up?

There's got to be something we can link to or add to the doc here to disambiguate this.  Even if we don't know what all the valid values are, we could say something like "This field is known as (foo) when using (some particular AWS API/command)".

If you're not sure, could you tell me:

- How do we end up using the field?  Do we end up giving it as input to some AWS API?
- Could you tell me how, using either AWS CLI or AWS console, I can get AWS to show me one of these values.

Reply:
product_type is used for the EntityType on function calls using Boto3. 
for describe an entity such as here:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/marketplace-catalog.html?highlight=awsmarketplace#MarketplaceCatalog.Client.describe_entity

or when creating a changeset like here:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/marketplace-catalog.html?highlight=awsmarketplace#MarketplaceCatalog.Client.start_change_set

You can look at the entities we already use by using list_entities here:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/marketplace-catalog.html?highlight=awsmarketplace#MarketplaceCatalog.Client.list_entities

You can also navigate to the Marketplace and look at the calls when navigating the Product drop down. They will look somewhat like this

https://aws.amazon.com/marketplace/management/products/api/v2/products?entityType=ServerProduct&maxResults=20&sortBy=LastModifiedDate&sortOrder=

There is the ability to use it with Offer but I couldn't find much on the usage of that. It's not great but it's all I've found so far.

Reply:
Thanks.  I wasn't able to find 'products' API calls myself, maybe we are getting different views of the marketplace, but in any case the usage as EntityType seems like enough info.

If I'm getting this right, then valid values for this field would be only whatever values for EntityType are accepted by the Marketplace Catalog API.

If so, I don't think we should give this a generic name like `product_type`.  Other code might be interested in a kind of "product type" which is unrelated to the marketplace catalog and which can't use the same values.  In this case, it seems like it'd be better to name the field more precisely around how it's used.  If values are specific to this one AWS API then it can't really be used for anything else.

So, may I suggest a better name for this would be something like `marketplace_entity_type` with a doc string mentioning something like "Entity type, as used by the AWS Marketplace Catalog API" (and with the examples retained).

Reply:
That makes sense, I've been mulling over changing this to entity_type for awhile but marketplace_entity_type is more precise.

Reply:
updated this one.